carapace-server/src/main/java/org/carapaceproxy/server/config/RequestMatcher.java

```diff
@@ -27,4 +27,6 @@
 public interface RequestMatcher {
 
     RoutingKey matches(HttpRequest request);
+
+    String appliedPattern();
```

Comment:
I think this is not applicable for every 'RequestMatcher', think about, we will have 'AndRequestMatcher', 'OrRequestMatcher', 'NegateRequestMatcher'

I suggest you to have a 'getDescription()' with a javadoc which explains that this is a description for the UI
this way every Matcher will create a suitable message